Language/Haskell/GhcMod/ImportedFrom.hs

```diff
@@ -0,0 +1,826 @@
+--                  }
+-- , NiceImportDecl {modName = "Safe"
+--                  , modQualifier = Nothing
+--                  , modIsImplicit = False
+--                  , modHiding = []
+--                  , modImportedAs = Nothing
+--                  , modSpecifically = []
+--                  }
+-- , NiceImportDecl { modName = "System.Environment"
+--                  , modQualifier = Nothing
+--                  , modIsImplicit = False
+--                  , modHiding = []
+--                  , modImportedAs = Nothing
+--                  , modSpecifically = ["getArgs"]
+--                  }
+-- , NiceImportDecl { modName = "Data.List"
+--                  , modQualifier = Nothing
+--                  , modIsImplicit = False
+--                  , modHiding = ["map"]
+--                  , modImportedAs = Nothing
+--                  , modSpecifically = []
+--                  }
+-- ]
+toImportDecl :: GHC.DynFlags -> SrcLoc.Located (GHC.ImportDecl GHC.RdrName) -> NiceImportDecl
+toImportDecl dflags idecl = NiceImportDecl
+                            { modName           = name
+                            , modQualifier      = qualifier
+                            , modIsImplicit     = isImplicit
+                            , modHiding         = hiding
+                            , modImportedAs     = importedAs
+                            , modSpecifically   = specifically
+                            }
+  where
+    idecl'       = SrcLoc.unLoc idecl
+    name         = showSDoc dflags (ppr $ GHC.ideclName idecl')
+    isImplicit   = GHC.ideclImplicit idecl'
+    qualifier    = unpackFS <$> ghc_sl_fs <$> GHC.ideclPkgQual idecl'
+    hiding       = (catMaybes . parseHiding . ghcIdeclHiding) idecl'
+    importedAs   = (showSDoc dflags . ppr) <$> ideclAs idecl'
+    specifically = (parseSpecifically . ghcIdeclHiding) idecl'
+
+    grabNames :: GHC.Located [GHC.LIE GHC.RdrName] -> [String]
+    grabNames loc = map (showSDoc dflags . ppr) names
+      where names :: [RdrName]
+            names = map (ieName . SrcLoc.unLoc) $ SrcLoc.unLoc loc
+            -- FIXME We are throwing away location info by using unLoc each time?
+            -- Trace these things to see what we are losing.
+
+    parseHiding :: Maybe (Bool, Located [LIE RdrName]) -> [Maybe String]
+    parseHiding Nothing = [Nothing]
+
+    -- If we do
+    --
+    --     import System.Environment ( getArgs )
+    --
+    -- then we get ["getArgs"] here, but we don't really need it...
+    parseHiding (Just (False, _)) = []
+
+    -- Actually hid names, e.g.
+    --
+    --     import Data.List hiding (map)
+    parseHiding (Just (True, h))  = map Just $ grabNames h
+
+    parseSpecifically :: Maybe (Bool, Located [LIE RdrName]) -> [String]
+    parseSpecifically (Just (False, h)) = grabNames h
+    parseSpecifically _                 = []
+
+-- This definition of separateBy is taken
+-- from: http://stackoverflow.com/a/4978733
+separateBy :: Eq a => a -> [a] -> [[a]]
```

Comment:
Why not use `splitOn` instead? I'm not sure what license stackoverflow uses for code examples. I think it's CC-SA which is probably going to conflict with the GPL, better not just plain copy stuff from there [until they switch to MIT.](https://meta.stackexchange.com/questions/272956/a-new-code-license-the-mit-this-time-with-attribution-required?cb=1)